src/stream/src/common/table/state_table.rs

```diff
@@ -714,6 +774,10 @@ where
         let values = value_chunk.serialize_with(&self.row_serde);
 
         let key_chunk = chunk.reorder_columns(self.pk_indices());
+        // TODO(kwannoel): Seems like we are doing vis check twice here.
+        // Once below, when using vis, and once here,
+        // when using vis to set rows empty or not.
+        // If we are to use the vis optimization, we should skip this.
```

Comment:
https://github.com/risingwavelabs/risingwave/issues/11377

Refactor it after this PR.